src/manifolds/Stiefel.jl

```diff
@@ -287,6 +373,19 @@ where $\operatorname{sgn}(p) = \begin{cases}
 """
 retract(::Stiefel, ::Any, ::Any, ::QRRetraction)
 
+function retract!(::Stiefel, q, p, X, ::PadeRetraction{m}) where {m}
+    Pp = I - 1 // 2 * p * p'
+    WpX = Pp * X * p' - p * X' * Pp
+    pm = sum([
```

Comment:
I think a simple loop would be the fastest solution here. Something like
```julia
pm = similar(WpX) # to make sure that `pm` is mutable
copyto!(pm, I)
qm = similar(WpX)
copyto!(qm, I)
WpXk = similar(WpX)
copyto!(WpXk, factorial(m) / factorial(2m) * I)
for k in 0:m
    WpXk .*= (some simplification of these factorials)
    pm .+= WpXk
    if k%2 == 0
        qm .+= WpXk
    else
        qm .-= WpXk
    end
    WpXk *= WpX
end
```

Reply:
Thanks for the sketch, but I am not really able to figure out the simplifications of the factor. To me it looks like it should be
```
WpXk  .*= ( k==0 ? 1 : (m-k+1)/( (2*m-k+1) * k) )
```
the first “removes” that factor from the 1/(m-k)! (namely the (m-(k-1)) factor) the second the similar term from the nominator and we have to divide by k for building k!. However, this does not yield the same results as before.

Reply:
Found the errors. For k=0 it has to be 2, and both `pm`, `qm` have to be initialised to be zero (not `I`). Now it works, thanks for the sketch. This part was actually one area where I hoped for an optimization.